solutions/devsprint-julio-fernandes-2/FinanceAppTests/Service/FinanceServiceTests.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  FinanceServiceTests.swift
+//  FinanceAppTests
+//
+//  Created by Thiago Cezario on 17/08/22.
+//
+
+import XCTest
+@testable import FinanceApp
+
+final class FinanceServiceTests: XCTestCase {
+    var url: URL!
+    
+    override func setUp() {
+        url = URL(string: "https://www.testURL.com")!
+    }
+    
+    override func tearDown() {
+        url = nil
+    }
+    
+    func testFetchActivityDetailsSuccess() {
+        let networkClient = NetworkClientMock()
+        
+        networkClient.performRequest(with: url) { data in
+            guard let data = data else {
+                XCTFail()
+                return
+            }
+            
+            let decoder = JSONDecoder()
+            decoder.keyDecodingStrategy = .convertFromSnakeCase
+            guard let activityDetails = try? decoder.decode(ActivityDetails.self, from: data) else {
+                XCTFail()
+                return
+            }
+            
+            XCTAssertNotNil(activityDetails)
+        }
+    }
```

Comment:
Tudo é apenas sugestão hahaha [aqui tem um artigo](https://medium.com/@bruno.hcr/swift-tests-tips-tricks-why-shouldnt-you-use-force-unwrapping-in-your-test-suite-29c7e973a70b) que fala sobre o porque de não utilizar forceunwrapp nos testes..

Até mesmo nos guad let, acho legal usar o XCTUnwrap

Reply:
Eu particularmente em relaçao a testes não vejo problema nenhum, é um ambiente controlado sem variaçoes, entao se aparecer um nil você vai ficar sabendo